### PR TITLE
Lua Number -> Double; add bitOp

### PR DESCRIPTION
**Pull request by brentpicasso**
* change Lua number to double to fix 29 bit CAN ID overflow and other overflow issues
* add bitOp library

### CODE REVIEW

File: lib_lua/src/luaconf.h [hunk 1]
```diff
@@ -506,9 +506,11 @@
 #if defined(LUA_NUMBER_INTEGER)
 #define LUA_NUMBER      int
 #else
-#define LUA_NUMBER      float
+#define LUA_NUMBER      double
 #endif
 
+#define LUA_NUMBER_DOUBLE LUA_NUMBER
```
Comment (stieg) [inline lib_lua/src/luaconf.h]: This is so weird.  But whatever.

**Comment by stieg** [comment]
:shipit:

**Comment by kosenko-max** [comment]
Could be the reason for elevated memory usage.

**Comment by stieg** [comment]
Yeah.  Needed doubles for bitops and 29bit CAN messages.

On Sun, Feb 14, 2016 at 1:28 PM, Max <notifications@github.com> wrote:

> Could be the reason for elevated memory usage.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/392#issuecomment-183960164>
> .
>



--

**Pull request merged**